tarogarden/seedling.go

```diff
@@ -124,6 +128,49 @@ func (c Seedling) validateFields() error {
 	return nil
 }
 
+// validateGroupKey attempts to validate that the non-zero group key provided
+// with a seedling is owned by the daemon and can be used with this seedling.
+func (c Seedling) validateGroupKey(groups []*asset.AssetGroup) (
+	*asset.AssetGroup, error) {
+
+	if len(groups) == 0 {
+		return nil, fmt.Errorf("group key %v not found",
+			c.GroupInfo.GroupKey.GroupPubKey.SerializeCompressed())
+	}
+
+	// Asset groups are sorted in insertion order, so the first group should
+	// include the information used for the original tweak.
+	seedlingGroup := groups[0]
+
+	// We must be able to sign with the group key.
+	if !seedlingGroup.GroupKey.IsLocal() {
+		return nil, fmt.Errorf("cannot sign with group key %v",
+			c.GroupInfo.GroupKey.GroupPubKey.SerializeCompressed())
+	}
+
+	// The seedling asset type must match the group asset type.
+	if c.AssetType != seedlingGroup.Genesis.Type {
```

Comment:
don't we need `meta`, `tag` to match also?

Reply:
I don't think so - looking at the main BIP, the type isn't explicitly required to match, but is part of the original input for the key tweak, so in practice must match (each asset tree is of only one type).

I can imagine there are use cases for different `meta` or `tag` values over multiple tranches as well.

Reply:
> don't we need meta, tag to match also?

Those were bound to the _original_ assets, while a new asset can specify these values anew. They _can_ and in most cases will be the same, tho.